src/Univ2LpOracle.sol

```diff
@@ -247,7 +249,26 @@ contract UNIV2LPOracle {
 
         // Get LP token supply
         uint256 supply = ERC20Like(src).totalSupply();
-        require(supply != 0, "UNIV2LPOracle/invalid-lp-token-supply");
+        require(supply > 0, "UNIV2LPOracle/invalid-lp-token-supply");
+
+        // Adjust LP token supply if fees are on
+        // We sacrifice some correctness for gas-savings by not querying param `feeTo`
+        // from the factory. This is only an issue during the time interval that Uniswap
+        // LP fees are toggled on/off and liquidity in the pair is added/removed.
+        // Uniswap Governance toggling fees is assumed to be a rare event, with liquidity
+        // for popular pairs constantly in flux. In the worst case the LP price would be off
+        // by ~0.05% which is insigificant relative to drift caused by the `hop` delay.
+        uint256 kLast = UniswapV2PairLike(src).kLast();
+        if (kLast > 0) {
+            uint256 rootK = sqrt(k);
+            uint256 rootKLast = sqrt(kLast);
+            if (rootK > rootKLast) {
+                uint256 numerator = wmul(supply, sub(rootK, rootKLast));
```

Comment:
should we expand these to long-form like we with the price quote calculation in line 274?
I think it looks cleaner without, but we may want to be consistent.

Reply:
Personally, I'm not bothered by the way things are. There are fewer operations in this expression so there's less reason to expand it.

Reply:
Also you can do a gas & precision optimization: change `wmul` to `mul` here, and then on line 268 change `wdiv` to just `/`.

Reply:
fixed